### PR TITLE
Add null check on originalEncoding parameter of EncodingModel

### LINKED ISSUE #7057: Cannot create Ant Freeform project
### Apache NetBeans version

Apache NetBeans 20

### What happened

I wanted to create a freeform project from an existing Ant `build.xml` script. 

It's impossible to proceed past the wizard stage where you map the project targets (build, clean etc) to the ant script. There is an exception shown in the status line.

### Language / Project Type / NetBeans Component

Java freeform project

### How to reproduce

1. Have a pre-existing non-netbeans ant project
2. In netbeans,  select "New Project | Java with Ant | Java Free-form project"
3. Press next
4. Choose the dir with build.xml in it
5. Pick a project name
6. Press next
7. No need to change the default build an run actions
8. Press next

The wizard does not advance to the next step. An exception notification appears on the bottom of the main IDE screen

### Did this work correctly in an earlier version?

It certainly did at some point but haven't checked in the last (approx) 5 versions

### Operating System

Windows 11 version 10.0 running on amd64; UTF-8; en_GB (nb)

### JDK

21.0.1; OpenJDK 64-Bit Server VM 21.0.1+12-LTS

### Apache NetBeans packaging

Apache NetBeans provided installer

### Anything else

It is reproducible.

I can look further into it, not sure I could make a PR though.

### Are you willing to submit a pull request?

No

### Exception details
```
java.lang.NullPointerException: Cannot invoke "Object.equals(Object)" because "o" is null
	at java.base/java.util.ImmutableCollections$Set12.contains(ImmutableCollections.java:817)
	at org.netbeans.spi.project.ui.support.ProjectCustomizer$EncodingModel.<init>(ProjectCustomizer.java:755)
	at org.netbeans.spi.project.ui.support.ProjectCustomizer.encodingModel(ProjectCustomizer.java:728)
	at org.netbeans.modules.java.freeform.ui.SourceFoldersPanel.updateEncodingCombo(SourceFoldersPanel.java:164)
	at org.netbeans.modules.java.freeform.ui.SourceFoldersPanel.setModel(SourceFoldersPanel.java:851)
	at org.netbeans.modules.java.freeform.ui.SourceFoldersWizardPanel.readSettings(SourceFoldersWizardPanel.java:138)
	at org.openide.WizardDescriptor.updateStateOpen(WizardDescriptor.java:843)
	at org.openide.WizardDescriptor.updateState(WizardDescriptor.java:797)
	at org.openide.loaders.TemplateWizard.updateState(TemplateWizard.java:758)
	at org.netbeans.modules.project.ui.NewProjectWizard.updateState(NewProjectWizard.java:59)
	at org.openide.WizardDescriptor._updateState(WizardDescriptor.java:775)
	at org.openide.WizardDescriptor.goToNextStep(WizardDescriptor.java:1041)
	at org.openide.WizardDescriptor.access$1900(WizardDescriptor.java:67)
	at org.openide.WizardDescriptor$Listener$1.run(WizardDescriptor.java:2184)
	at org.openide.WizardDescriptor.lazyValidate(WizardDescriptor.java:1492)
	at org.openide.WizardDescriptor.access$2000(WizardDescriptor.java:67)
	at org.openide.WizardDescriptor$Listener.actionPerformed(WizardDescriptor.java:2203)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.openide.util.WeakListenerImpl$ProxyListener.invoke(WeakListenerImpl.java:461)
	at jdk.proxy1/jdk.proxy1.$Proxy29.actionPerformed(Unknown Source)
	at java.desktop/javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:1972)
	at java.desktop/javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2314)
	at java.desktop/javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:407)
	at java.desktop/javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:262)
	at java.desktop/javax.swing.plaf.basic.BasicButtonListener.mouseReleased(BasicButtonListener.java:279)
	at java.desktop/java.awt.Component.processMouseEvent(Component.java:6621)
	at java.desktop/javax.swing.JComponent.processMouseEvent(JComponent.java:3398)
	at java.desktop/java.awt.Component.processEvent(Component.java:6386)
	at java.desktop/java.awt.Container.processEvent(Container.java:2266)
	at java.desktop/java.awt.Component.dispatchEventImpl(Component.java:4996)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2324)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4828)
	at java.desktop/java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4948)
	at java.desktop/java.awt.LightweightDispatcher.processMouseEvent(Container.java:4575)
	at java.desktop/java.awt.LightweightDispatcher.dispatchEvent(Container.java:4516)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2310)
	at java.desktop/java.awt.Window.dispatchEventImpl(Window.java:2780)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4828)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:775)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:720)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:714)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:400)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:87)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:98)
	at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:747)
	at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:745)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:400)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:87)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:744)
	at org.netbeans.core.TimableEventQueue.dispatchEvent(TimableEventQueue.java:136)
[catch] at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:117)
	at java.desktop/java.awt.WaitDispatchSupport$2.run(WaitDispatchSupport.java:191)
	at java.desktop/java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:236)
	at java.desktop/java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:234)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:319)
	at java.desktop/java.awt.WaitDispatchSupport.enter(WaitDispatchSupport.java:234)
	at java.desktop/java.awt.Dialog.show(Dialog.java:1079)
	at org.netbeans.core.windows.services.NbPresenter.superShow(NbPresenter.java:1069)
	at org.netbeans.core.windows.services.NbPresenter.doShow(NbPresenter.java:1114)
	at org.netbeans.core.windows.services.NbPresenter.run(NbPresenter.java:1092)
	at org.netbeans.core.windows.services.NbPresenter.run(NbPresenter.java:104)
	at org.netbeans.modules.openide.util.NbMutexEventProvider$Event.doEventAccess(NbMutexEventProvider.java:123)
	at org.netbeans.modules.openide.util.NbMutexEventProvider$Event.readAccess(NbMutexEventProvider.java:77)
	at org.netbeans.modules.openide.util.LazyMutexImplementation.readAccess(LazyMutexImplementation.java:71)
	at org.openide.util.Mutex.readAccess(Mutex.java:199)
	at org.netbeans.core.windows.services.NbPresenter.show(NbPresenter.java:1076)
	at java.desktop/java.awt.Component.show(Component.java:1728)
	at java.desktop/java.awt.Component.setVisible(Component.java:1675)
	at java.desktop/java.awt.Window.setVisible(Window.java:1036)
	at java.desktop/java.awt.Dialog.setVisible(Dialog.java:1015)
	at org.netbeans.core.windows.services.DialogDisplayerImpl$AWTQuery.showDialog(DialogDisplayerImpl.java:273)
	at org.netbeans.core.windows.services.DialogDisplayerImpl$AWTQuery.run(DialogDisplayerImpl.java:188)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:318)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:773)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:720)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:714)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:400)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:87)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:742)
	at org.netbeans.core.TimableEventQueue.dispatchEvent(TimableEventQueue.java:136)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
```

### PR DESCRIPTION
**Pull request by pedro-w**
`EncodingModel` is an internal class, implementing `ComboBoxModel`, which is used for a combo box intended to show all the encodings available to the current JVM. It is constructed with a parameter called `originalEncoding` which pre-selects the encoding to show in the combo. A value of null means to pick the default encoding. 
Due to implementation changes, passing in null causes a null pointer exception when `Set#contains()` is called.
This PR adds an extra check to skip straight to selecting the default encoding if `originalEncoding` is null.

Fixes #7057

### CODE REVIEW

**Review by matthiasblaesing — APPROVED** [review]
Looks sane to me. Thank you. 

master currently fails in the LSP tests. Once that is resolved, I'll merge this.

**Comment by pedro-w** [comment]
I can add a test for it too, if you can wait a bit (or let me know if not
needed)

**Comment by matthiasblaesing** [comment]
@pedro-w sure can do :+1:

**Comment by pedro-w** [comment]
OK, done.

**Comment by mbien** [comment]
@pedro-w please squash and force push so that we can merge this whole PR as-is once master is green again.

**Review by mbien — APPROVED** [review]

**Comment by matthiasblaesing** [comment]
Thanks again.

**Pull request merged**
